### PR TITLE
Auto-generate version index file for NodeJS.

### PR DESCRIPTION
**Pull request by shinfan**
cc/ @swcloud @bjwatson

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
The generator changes LGTM.

File: src/main/resources/com/google/api/codegen/nodejs/version_index.snip [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@extends "nodejs/common.snip"
+
+@snippet generate(index)
+  /*
+   {@copyright()}
+   */
+  'use strict';
+
+  module.exports.v1 = require('./{@index.apiVersion}');
```
Comment (landrito) [inline src/main/resources/com/google/api/codegen/nodejs/version_index.snip]: This should be:

```
module.exports.{@index.apiVersion} = require('./{@index.apiVersion}');

Comment (shinfan) [inline src/main/resources/com/google/api/codegen/nodejs/version_index.snip]: Oops you are right. This one was not caught by the baseline. Fixed

**Comment by shinfan** [comment]
@landrito PTAL

File: src/test/java/com/google/api/codegen/testdata/nodejs_doc_version_index_library.baseline [hunk 2]
```diff
@@ -0,0 +1,19 @@
+============== file: src/index.js ==============
+/*
+ * Copyright 2016 Google Inc. All rights reserved.
```
Comment (landrito) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_version_index_library.baseline]: nit: Should we be using 2017 now?

Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_version_index_library.baseline]: Lol you are right. This is a from a shared component so I prefer to fix in a separate PR. https://github.com/googleapis/toolkit/issues/897

**Pull request merged**